pkgs/tools/misc/vrc-get/default.nix

```
@@ -22,4 +22,11 @@ rustPlatform.buildRustPackage rec {
     maintainers = with maintainers; [ bddvlpr ];
     mainProgram = "vrc-get";
   };
+
+  postInstall = ''
```

```suggestion
  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
```

We seem to missing tons of this...

---

Good point on cross-compilation compatibility, although perhaps it could emulate to still be able to generate completions most of the time.
```diff
@@ -1,4 +1,4 @@
-{ fetchCrate, installShellFiles, lib, rustPlatform, pkg-config, stdenv, Security, SystemConfiguration }:
+{ fetchCrate, installShellFiles, lib, rustPlatform, pkg-config, stdenv, Security, SystemConfiguration, buildPackages }:
 
 rustPlatform.buildRustPackage rec {
   pname = "vrc-get";
@@ -23,10 +23,10 @@ rustPlatform.buildRustPackage rec {
     mainProgram = "vrc-get";
   };
 
-  postInstall = ''
+  postInstall = lib.optionalString (stdenv.buildPlatform.emulatorAvailable buildPackages) ''
     installShellCompletion --cmd vrc-get \
-      --bash <($out/bin/vrc-get completion bash) \
-      --fish <($out/bin/vrc-get completion fish) \
-      --zsh <($out/bin/vrc-get completion zsh)
+      --bash <(${stdenv.hostPlatform.emulator buildPackages} $out/bin/vrc-get completion bash) \
+      --fish <(${stdenv.hostPlatform.emulator buildPackages} $out/bin/vrc-get completion fish) \
+      --zsh <(${stdenv.hostPlatform.emulator buildPackages} $out/bin/vrc-get completion zsh)
   '';
 }
```

The other idea I'm thinking is that when cross-compiling it could depend on the native version of itself for completions. That should always work.
```diff
@@ -1,4 +1,4 @@
-{ fetchCrate, installShellFiles, lib, rustPlatform, pkg-config, stdenv, Security, SystemConfiguration }:
+{ fetchCrate, installShellFiles, lib, rustPlatform, pkg-config, stdenv, Security, SystemConfiguration, buildPackages }:
 
 rustPlatform.buildRustPackage rec {
   pname = "vrc-get";
@@ -9,7 +9,7 @@ rustPlatform.buildRustPackage rec {
     hash = "sha256-+xbHw1DpFmapjsFoUvxUqTok8TKMebMw3gYjO/rx/iU=";
   };
 
-  nativeBuildInputs = [ installShellFiles pkg-config ];
+  nativeBuildInputs = [ installShellFiles pkg-config ] ++ lib.optional (!stdenv.buildPlatform.canExecute stdenv.hostPlatform) buildPackages.vrc-get;
 
   buildInputs = lib.optionals stdenv.isDarwin [ Security SystemConfiguration ];
 
@@ -23,10 +23,12 @@ rustPlatform.buildRustPackage rec {
     mainProgram = "vrc-get";
   };
 
-  postInstall = ''
-    installShellCompletion --cmd vrc-get \
-      --bash <($out/bin/vrc-get completion bash) \
-      --fish <($out/bin/vrc-get completion fish) \
-      --zsh <($out/bin/vrc-get completion zsh)
-  '';
+  postInstall =
+    let bin = if (stdenv.buildPlatform.canExecute stdenv.hostPlatform) then "$out/bin/vrc-get" else "${buildPackages.vrc-get}/bin/vrc-get";
+    in ''
+      installShellCompletion --cmd vrc-get \
+        --bash <(${bin} completion bash) \
+        --fish <(${bin} completion fish) \
+        --zsh <(${bin} completion zsh)
+    '';
 }
```